django/contrib/contenttypes/fields.py

```diff
@@ -359,10 +359,43 @@ def resolve_related_fields(self):
         self.to_fields = [self.model._meta.pk.name]
         return [(self.remote_field.model._meta.get_field(self.object_id_field_name), self.model._meta.pk)]
 
+    def _get_path_info_with_parent(self):
+        """
+        Returns the path with joining through the parent model.
+        """
+        # In a situation where you have inheritance chain ChildTag -> Tag,
+        # and Tagged defines the GenericForeignKey, and a TaggedItem model has a
```

Comment:
Yes. The problem is that if you have TaggedItem and it has a relation to ChildTag, then currently (before this PR) a filter against the childtag joins directly from TaggedItem to ChildTag. But the needed fields for the join condition (object_id and content_type) are on the Tag model.